src/plugins/unified_search/public/query_string_input/query_string_input.tsx

```diff
@@ -352,6 +366,28 @@ export default class QueryStringInputUI extends PureComponent<QueryStringInputPr
     }
   };
 
+  private validateKql = (query: string) => {
+    if (
+      this.props.validateKqlQuery &&
+      this.props.query.language === 'kuery' &&
+      this.state.dataViews.length > 0
```

Comment:
we could validate even if the dataview is not specified, it would be just less optimal right?